### PR TITLE
kx122: Initial implementation

### PR DESCRIPTION
**Pull request by RissanenGit**
The driver implements support for Kionix's KX122 accelerometer.
The KX122 is a tri-axis ±2g, ±4g or ±8g accelerometer with integrated 2048 byte buffer, orientation, tap/double tap, activity detecting, and Free fall algorithms.

The current version of the driver implements functionality for the accelerometer and it's configuration, 
buffer's functionality and it's configuration and interrupt functionalities.

The driver was tested on Raspberry Pi 3 model b

Signed-off-by: Samuli Rissanen <samuli.rissanen@hotmail.com>

### CODE REVIEW

File: src/kx122/kx122.hpp [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+      */
+      float getSamplePeriod();
+
+      /**
+      Gets who am i value of the sensor.
+
+      @param data Pointer to a uint8_t variable to store the value.
+      @throws std::runtime_error on failure.
+      */
+      void getWhoAmI(uint8_t *data);
```
Comment (pylbert) [inline src/kx122/kx122.hpp]: Double-check this - I'm not seeing the implementation for getWhoAmI anywhere.

Comment (RissanenGit) [reply src/kx122/kx122.hpp]: Interesting that i somehow managed to completely miss that.
Anyways, the implementation is now added.

**Comment by RissanenGit** [comment]
Out of the other languages, the only one i have used is Python.
I could try to whip up an example for that.

I assume that the Python module i can use for the example is automatically created from the C++ wrapper?

**Comment by pylbert** [comment]
Nice addition Samuli - merged.

Note, I added the other language examples and an stl::vector method for getting the acceleration.

**Pull request closed without merging**